OsmIntegrator/Services/OsmUpdater.cs

```diff
@@ -344,5 +366,43 @@ private void UpdateStopProperties(DbStop stop, List<Tag> newTags, ReportStop rep
         stop.Number = localRefTag.Value;
       }
     }
+
+    public async Task UpdateTileReferences(List<DbTile> tiles, ApplicationDbContext dbContext)
+    {
+      using (IDbContextTransaction transaction = await dbContext.Database.BeginTransactionAsync())
+      {
+
+        try
+        {
+          foreach (DbTile tile in tiles)
+          {
+            foreach (DbStop stop in tile.Stops)
+            {
+              double stopLat = stop.Lat;
+              double stopLon = stop.Lon;
+
+              if (!(stopLat >= tile.MinLat && stopLat <= tile.MaxLat && stopLon >= tile.MinLon && stopLon <= tile.MaxLon))
+              {
+                stop.Tile = dbContext.Tiles.FirstOrDefault(tile =>
+                  stopLat >= tile.MinLat &&
```

Comment:
Lepiej by było sprawdzić numer kafelka funkcją `TilesHelper.WorldToTilePos(stop.Lon, stop.Lat, _zoomLevel)`. Dzięki temu jeżeli przystanek ZTM przesunął się poza istniejące kafelki to będziemy mogli dodać taki kafelek w przyszłości.
Ten sposób został wykorzystany w metodzie `  private List<DbTile> GetTiles(List<DbStop> stops)` w klasie `DataInitializer`.

Reply:
Poprawione